File: ShapesPainter/app/src/main/java/com/example/alexander/shapespainter/ICanvas.java
package com.example.alexander.shapespainter;

import android.graphics.Canvas;

import javax.vecmath.Vector2f;


public interface ICanvas {
    void drawRectangle(Vector2f leftTop,
                       Vector2f topBottom,
                       Canvas canvas);

    void drawEllipse(Vector2f center,
                     float hRadius,
                     float vRadius,
                     Canvas canvas);

    void drawTriangle(Vector2f vertex1,
                      Vector2f vertex2,
                      Vector2f vertex3,
                      Canvas canvas);
}

File: Fabric/app/src/main/java/com/example/alexander/fabric/WorkWithShapes/ICanvas.kt
package com.example.alexander.fabric.WorkWithShapes

import com.example.alexander.fabric.Definitions.ColorEnum
import com.example.alexander.fabric.Definitions.CordPoint

/**
 * Created by Alexander on 25.06.2016.
 */
interface ICanvas
{
    fun SetColor(color: ColorEnum)
    fun DrawLine(from: CordPoint, to: CordPoint)
    fun DrawEllipse(center: CordPoint, hRadius: Float, vRadius: Float)
}